lib/middleware/backbone_request_id.js

```diff
@@ -0,0 +1,11 @@
+import Backbone from 'backbone'
+import { wrap } from 'underscore'
+
+module.exports = (req, res, next) => {
+  Backbone.sync = wrap(Backbone.sync, (sync, method, model, options = {}) => {
```

Comment:
Unfortunately, this seems like a possibly not great solution. `Backbone.sync` is a global request layer, so by doing this you would be swapping out `Backbone.sync` thousands of times per minute and maybe from underneath in-flight requests. Unfortunately, I think the only way to tie the express request to the Backbone-level fetch is like you did with `metaphysics` above where you pass the request info through via `model.fetch(data : headers: 'X-Request-Id': req.get('X-Request-Id')`